src/model/document/document.ts

```diff
@@ -206,6 +208,45 @@ function useElementFocus() {
   }
 }
 
+function useElementDrag() {
+  function makeDraggable(element: QuantumElement, gridCellSize: Vector2) {
+    // wait until the element gets created in the DOM
+    nextTick(function () {
+      var DOMElement = document.getElementById(element.id)
+      if (DOMElement)
+        interact(DOMElement)
+          .draggable({
+            ignoreFrom: '.quantum-element',
+            modifiers: [
+              interact.modifiers.snap({
+                targets: [interact.snappers.grid({ x: 20, y: 20 })],
+                range: Infinity,
+                relativePoints: [{ x: 0, y: 0 }],
+              }),
+              interact.modifiers.restrict({
+                restriction: DOMElement.parentNode as HTMLElement,
+                elementRect: { top: 0, left: 0, bottom: 1, right: 1 },
+                endOnly: true,
+              }),
+            ],
+            inertia: false,
+          })
+          .on('down', function (event) {
```

Comment:
Feel free to use a lambda, as in `(event) => {` instead of `function (event) {`
However, seeing as how this is a purely stylistic choice, I don't mind either way.

Reply:
I do like arrow functions too, will change these over